server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceRegistry.java

```diff
@@ -150,14 +152,38 @@ public ValuesSourceRegistry build() {
 
     /** Maps Aggregation names to (ValuesSourceType, Supplier) pairs, keyed by ValuesSourceType */
     private final AggregationUsageService usageService;
-    private Map<RegistryKey<?>, Map<ValuesSourceType, ?>> aggregatorRegistry;
+    private final Map<RegistryKey<?>, Map<ValuesSourceType, ?>> aggregatorRegistry;
+    private final Map<ValuesSourceType, Set<String>> supportedAggsByValueSourceType;
 
     public ValuesSourceRegistry(
         Map<RegistryKey<?>, List<Map.Entry<ValuesSourceType, ?>>> aggregatorRegistry,
         AggregationUsageService usageService
     ) {
         this.aggregatorRegistry = copyMap(aggregatorRegistry);
         this.usageService = usageService;
+
+        // First find all unique values source types:
+        Set<ValuesSourceType> supportedTypes = new HashSet<>();
+        for (var entryList : aggregatorRegistry.values()) {
+            for (var entry : entryList) {
+                supportedTypes.add(entry.getKey());
+            }
+        }
+        // Per value source type create an immutable set of supported aggregations:
+        Map<ValuesSourceType, Set<String>> supportedAggsByValueSourceType = new HashMap<>();
+        for (ValuesSourceType supportedType : supportedTypes) {
+            Set<String> supportedAggs = new HashSet<>();
+            for (var entry : aggregatorRegistry.entrySet()) {
```

Comment:
I think that not all aggregations use the `ValuesSourceRegistry`. So not all aggregations have registered an aggregatorSupplier on a per values source type basis yet. We need to check which aggregations are not using `ValuesSourceRegistry` and make these aggregations use it.

For counter value source type, it looks that all supported aggregations use `ValuesSourceRegistry`.